src/components/Inputs/Select/Select.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { useRef, useState } from 'react';
+import { useOnOutsideClick } from '../../../hooks';
+import { ChevronDown, ChevronUp } from '../../../icons/regular';
+
+import { ColorsMap, SelectOption, SelectProps } from '../types';
+
+interface SelectSpecificProps extends SelectProps {
+  leftPadding?: string;
+  hideArrow?: boolean;
+}
+
+const Select = ({
+  searchable = false,
+  options,
+  defaultValue,
+  size = 'large',
+  state = 'idle',
+  placeholder = '',
+  onChange,
+  leftPadding = 'pl-0',
+  hideArrow = false,
+}: SelectSpecificProps) => {
+  const [selectedOption, setSelectedOption] = useState(defaultValue);
+  const [open, setOpen] = useState(false);
+  const [searchResults, setSearchResults] = useState(options);
+  const isDisabled = state === 'disabled';
+
+  const height = size === 'small' ? 'h-8' : 'h-11';
+  const textSize = size === 'small' ? 'text-sm' : 'text-base';
+  const iconSize = size === 'large' ? 'w-5 h-5' : 'w-4 h-4';
+
+  const idleBorderColor = selectedOption
+    ? 'border-dark-400'
+    : 'border-dark-300';
+  const openBorderColor = 'border-primary-1000';
+  const colors: ColorsMap = {
+    idle: `text-dark-1000 ${open ? openBorderColor : idleBorderColor} `,
+    error: 'text-dark-1000 border-error-1000',
+    success: 'text-dark-1000 border-green-1000',
+    disabled: 'text-dark-600 border-dark-300 bg-dark-200',
+  };
+
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const selectRef = useRef<HTMLInputElement>(null);
+  useOnOutsideClick(selectRef, () => setOpen(false));
+
+  const handleOpenClose = () => {
+    const wasOpen = open;
+    setOpen(!open);
+    if (searchable && !wasOpen) searchInputRef?.current?.focus();
+    if (searchable && wasOpen) searchInputRef?.current?.blur();
+  };
+
+  const handleChange = (option: SelectOption) => {
+    setSelectedOption(option);
+    setOpen(false);
+    onChange(option);
+  };
+
+  const handleSearch = (e: any) => {
+    setSelectedOption(e.target.value);
+    setOpen(true);
+
+    setSearchResults(
+      options.filter((option: SelectOption) => {
+        const matchValue = option.value
+          .toLowerCase()
+          .includes(e.target.value.toLowerCase());
+        const matchLabel = option.label
+          .toLowerCase()
+          .includes(e.target.value.toLowerCase());
+        return matchValue || matchLabel;
+      }),
+    );
+  };
+
+  return (
+    <section ref={selectRef}>
+      <section
+        className={`flex items-center  w-full border pl-3 pr-2  ${height}  ${
+          colors[state]
+        } 
+        ${open ? 'rounded-t' : 'rounded'}
```

Comment:
We have installed a library for conditional classes
classNames({ 'rounded-t': open })